### PR TITLE
Add dotnet8 feeds in main nuget.config

### PR DESCRIPTION
**Pull request by carlossanlop**
Contributes to https://github.com/dotnet/runtime/issues/63375
Contributes to https://github.com/dotnet/runtime/issues/72112

### CODE REVIEW

**Comment by carlossanlop** [comment]
This PR is fixing the following item from https://github.com/dotnet/runtime/issues/72112:

> Add the dotnet8 feeds to the NuGet.config

But I think I have to bring back the dotnet7 feeds because of the next item:

> At a later point when all dependencies are available in the .NET 8 feeds, remove the .NET 7 feeds.

**Review by smasher164 — APPROVED** [review]

File: NuGet.config [hunk 1]
```diff
@@ -19,6 +19,9 @@
     <!-- TODO: Remove dotnet7 feeds when dependencies publish into dotnet8 feeds: https://github.com/dotnet/runtime/issues/63375. -->
     <add key="dotnet7" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json" />
     <add key="dotnet7-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7-transport/nuget/v3/index.json" />
+    <!-- TODO: Remove dotnet8 feeds when dependencies publish into dotnet9 feeds: https://github.com/dotnet/runtime/issues/74100 -->
```
Comment (ViktorHofer) [inline NuGet.config:22]: That note isn't necessary. This will naturally happen in a year when we brand to .NET 9.

```suggestion
```

**Review by ViktorHofer — APPROVED** [review]
Aside from the one comment, LGTM.

**Comment by carlossanlop** [comment]
Test failures are unrelated:
- `System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback`: https://github.com/dotnet/runtime/pull/74101
- JIT: https://github.com/dotnet/runtime/issues/74125

**Pull request merged**